src/Plugin/LLModelProviders/Vllm.php

```diff
@@ -0,0 +1,263 @@
+      '#title' => $this->t('Connection timeout'),
+      '#description' => $this->t('Seconds to wait for connection with the LLM.'),
+      '#default_value' => $this->configuration['timeouts']['connect'],
+    ];
+
+    $form['timeouts']['wait'] = [
+      '#type' => 'number',
+      '#title' => $this->t('Wait timeout'),
+      '#description' => $this->t('Seconds to wait for response from LLM.'),
+      '#default_value' => $this->configuration['timeouts']['wait'],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateConfigurationForm(array &$form, FormStateInterface $form_state): void {
+    $values = $form_state->getValues();
+
+    if (filter_var($values['url'], FILTER_VALIDATE_URL) === FALSE) {
+      $form_state->setErrorByName('url', $this->t('Invalid URL.'));
+    }
+
+    $filter_options = [
+      'options' => [
+        'min_range' => 1,
+        'max_range' => 65535,
+      ],
+    ];
+    if (filter_var($values['port'], FILTER_VALIDATE_INT, $filter_options) === FALSE) {
+      $form_state->setErrorByName('port', $this->t('Invalid port range. Should be between 1 and 65535.'));
+    }
+
+    if ($values['auth']['username'] && !$values['auth']['password']) {
+      $form_state->setErrorByName('auth][password', $this->t('Password is required when username is provided.'));
+    }
+    if (!$values['auth']['username'] && $values['auth']['password']) {
+      $form_state->setErrorByName('auth][username', $this->t('Username is required when password is provided.'));
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state): void {
+    if (!$form_state->getErrors()) {
+      $values = $form_state->getValues();
+      $configuration = [
+        'url' => $values['url'],
+        'port' => $values['port'],
+        'auth' => [
+          'username' => $values['auth']['username'],
+          'password' => $values['auth']['password'],
+        ],
+        'timeouts' => [
+          'connect' => $values['timeouts']['connect'],
+          'wait' => $values['timeouts']['wait'],
+        ],
+      ];
+
+      $this->setConfiguration($configuration);
+    }
+
+    // Try to connect to VLLM to test the connection.
+    try {
+      $this->listModels();
+      $this->messenger()->addMessage('Successfully connected to VLLM');
+    }
+    catch (\Exception $exception) {
+      $this->messenger()->addMessage('Error communication with VLLM: ' . $exception->getMessage(), 'error');
+    }
+  }
+
+  /**
+   * Get a client.
+   *
+   * @return \Drupal\llm_services\Client\Vllm
+   *   Client to communicate with VLLM.
+   */
+  public function getClient(): ClientVllm {
+    return new ClientVllm(
+      url: $this->configuration['url'],
+      port: $this->configuration['port'],
+      client:  \Drupal::httpClient(),
+      username: $this->configuration['auth']['username'],
+      password: $this->configuration['auth']['password'],
```

Comment:
Will this be stored in drupal config files? If so consider using Drupal\Core\Site\Settings and store it in settings.local instead to avoid unintentional export